### PR TITLE
set default sidebar toggle based on settings

### LINKED ISSUE #574: Remove github-proxy
@keegancsmith suggested in https://github.com/sourcegraph/deploy-sourcegraph/issues/121#issuecomment-432207823 that it should be possible to remove github-proxy.

I don't know what that entails, so I'm filing this issue.

### PR DESCRIPTION
**Pull request by Numbers88s**
## 🤔  Description

Hide the sidebar on Repo View based on user configuration

## 🔖  Relevant tickets

Fixes [#574](https://github.com/sourcegraph/customer/issues/574)

## 🚀  GIF
_What gif best describes this PR or how it makes you feel?_

hide

### CODE REVIEW

**Review by efritz — APPROVED** [review]
That seemed easy.

**Comment by Numbers88s** [comment]
Yup 😄

File: schema/settings.schema.json [hunk 1]
```diff
@@ -312,6 +312,11 @@
         "$ref": "#/definitions/QuickLink"
       }
     },
+    "sidebarToggle": {
```
Comment (olafurpg) [inline schema/settings.schema.json:315]: It’s not self explanatory whether the sidebar is visible on true or false. What do you think about renaming the setting to “sidebarVisibleByDefault”? Or something in that direction, might be good to loop in @rrhyne for suggestions. There are also many sidebars in the product but this setting only affects the sidebar in the blob view.

Comment (rrhyne) [reply schema/settings.schema.json:315]: Agree with @olafurpg, this should definately be **fileSidebarVisibleByDefault**, as we also have a search page sidebar.

**Comment by macraig** [comment]
@Numbers88s that was ⚡️! I guess they'll be able to get it in December's release then?

File: schema/settings.schema.json (hunk 1 above)
Comment (Numbers88s) [reply schema/settings.schema.json:315]: Sounds good happy to update it.

Comment (Numbers88s) [reply schema/settings.schema.json:315]: Here is the PR updating it https://github.com/sourcegraph/sourcegraph/pull/28568 👍

**Comment by Numbers88s** [comment]
> @Numbers88s that was ⚡️! I guess they'll be able to get it in December's release then?

It could be, yeah. I'm not sure about the release cycle or when it will happen this month.

**Comment by macraig** [comment]
@Numbers88s oh I meant to ask if the work is finished and merged or if there's anything lacking. If it's shipped then it should be available in the next release (either on the 10th or 20th) :)  

cc @alexAtSourcegraph

**Comment by Numbers88s** [comment]
> 

@macraig Yes the work is finished and merged. It should be available for the 10th or 20th release. 👍

**Pull request merged**